MantidQt/MantidWidgets/src/WorkspacePresenter/WorkspacePresenter.cpp

```diff
@@ -0,0 +1,63 @@
+#include "MantidQtMantidWidgets/WorkspacePresenter/WorkspacePresenter.h"
+#include "MantidQtMantidWidgets/WorkspacePresenter/IWorkspaceDockView.h"
+#include "MantidQtMantidWidgets/WorkspacePresenter/IADSAdapter.h"
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+WorkspacePresenter::WorkspacePresenter(DockView_wptr view,
+                                       ADSAdapter_uptr adapter)
+    : m_view(view), m_adapter(std::move(adapter)) {
+  m_adapter->registerPresenter(m_view.lock()->getPresenterWeakPtr());
+}
+
+void WorkspacePresenter::notify(ADSNotifiable::Flag flag) {
+  switch (flag) {
+  case ADSNotifiable::Flag::WorkspaceLoaded:
+    break;
```

Comment:
Yes and here you would call your private methods that act according to the flag.